src/app/qgisapp.cpp

```diff
@@ -1033,6 +1033,8 @@ QgisApp::QgisApp( QSplashScreen *splash, bool restorePlugins, bool skipVersionCh
   connect( mBrowserWidget, &QgsBrowserDockWidget::connectionsChanged, this, &QgisApp::connectionsChanged );
   connect( mBrowserWidget, &QgsBrowserDockWidget::openFile, this, &QgisApp::openFile );
   connect( mBrowserWidget, &QgsBrowserDockWidget::handleDropUriList, this, &QgisApp::handleDropUriList );
+  connect( QgsApplication::dataItemProviderRegistry(), &QgsDataItemProviderRegistry::providerAdded, mBrowserModel, &QgsBrowserModel::addProvider );
+  connect( QgsApplication::dataItemProviderRegistry(), &QgsDataItemProviderRegistry::providerWillBeRemoved, mBrowserModel, &QgsBrowserModel::removeProvider );
```

Comment:
I think rather than just connecting to registry from outside of the model, QgsBrowserModel should connect to it by itself in the constructor. So it will work with all browser models automatically...